plugwise/helper.py

```diff
@@ -681,30 +727,33 @@ def _all_appliances(self) -> None:
                 appl.fw = self.smile_fw_version
                 appl.hw = self.smile_hw_version
 
-            self._appl_data[appl.dev_id] = {
-                "class": appl.pwclass,
-                "fw": appl.fw,
-                "hw": appl.hw,
+            # Don't show orphaned non-legacy thermostat-types.
+            if (
+                not self._smile_legacy
+                and appl.pwclass in THERMOSTAT_CLASSES
+                and appl.location is None
+            ):
+                continue
+
+            temp_dict = {
+                "appl_id": appl.dev_id,
+                "appl_data": {"dev_class": appl.pwclass},
+            }
+
+            for key, value in {
+                "firmware": appl.fw,
+                "hardware": appl.hw,
                 "location": appl.location,
                 "mac_address": appl.mac,
                 "model": appl.model,
                 "name": appl.name,
+                "zigbee_mac_address": appl.zigbee_mac,
                 "vendor": appl.v_name,
-            }
+            }.items():
+                if value is not None or key == "location":
```

Comment:
Very nice rewrite to prevent the old if statement, kudo's! 👍

Reply:
:)